test/Sentry.Tests/HubTests.cs

```diff
@@ -639,5 +639,26 @@ public void CaptureTransaction_AfterTransactionFinishes_ResetsTransactionOnScope
             // Assert
             hub.WithScope(scope => scope.Transaction.Should().BeNull());
         }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(2)]
+        public void Dispose_IsEnabled_SetToFalse(int disposeCount)
+        {
+            // Arrange
+            var hub = new Hub(new SentryOptions
+            {
+                Dsn = DsnSamples.ValidDsnWithSecret
+            });
+
+            // Act
+            for (var i = 0; i < disposeCount; i++)
+            {
+                hub.Dispose();
+            }
+
+            // Assert
+            hub.IsEnabled.Should().BeFalse();
```

Comment:
As discussed on our call:

```suggestion
            var hub = new Hub(new SentryOptions
            {
                Dsn = DsnSamples.ValidDsnWithSecret,
                
            });
            var client = Substibute.For<ISentryClient>();
            hub.BindClient(client);
            // Act
            for (var i = 0; i < disposeCount; i++)
            {
                hub.Dispose();
            }
            client.Received(1).Dispose();
            // Assert
            hub.IsEnabled.Should().BeFalse();
```

Another test would verify that `IsDispose` returns true at first, and after Dispose is called, it returns false